mt-kahypar/io/partitioning_output.cpp

```diff
@@ -763,7 +763,7 @@ namespace mt_kahypar::io {
     std::sort(internal_pins.begin(), internal_pins.end());
     std::sort(internal_degree.begin(), internal_degree.end());
 
-    auto square = [&](size_t x) { return x * x; };
+    auto square = [&](double x) { return x * x; };
```

Comment:
This is actually no UB since unsigned overflow is well defined, but of course produces wrong results^^